src/containerapp/azext_containerapp/_dapr_utils.py

```diff
@@ -0,0 +1,340 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from typing import Dict
+from azure.cli.core.azclierror import ValidationError
+from knack.log import get_logger
+from ._clients import ContainerAppClient, DaprComponentPreviewClient
+from ._models import (
+    DaprComponent as DaprComponentModel,
+    DaprMetadata as DaprMetadataModel,
+    DaprServiceComponentBinding as DaprServiceComponentBindingModel,
+)
+
+logger = get_logger(__name__)
+
+SERVICEBIND_METADATA_CREATEDBY_KEY = "SB_CREATED_BY"
+SERVICEBIND_METADATA_CREATEDBY_VALUE = "azext_containerapp_daprutils"
```

Comment:
Should this be more generic? Thinking what would happen if we added this easy init logic to the portal too... they would need to use the same value (or if they use a different one, we'd have to check for both)

Reply:
I'm assuming this metadata is non-functional. Something to just track which tooling created the binding? so if we want the same from portal, the UI will have to come up with its own metadata value there. /cc @shubham1172 right?

Reply:
@ahmelsayed this metadata is used in the `init` logic to figure out if the Dapr Component was created by the `dapr-component init` CLI command or by the users directly (via the existing portal experience or `dapr-component set` CLI command). This determines if a component should be over-written or not (in case it already exists).

@anthonychu when we are using the portal, I am thinking of two things:
1. Users clicking on init in a brand-new managed environment - in which case this metadata is unused.
2. Users clicking on init when some components already exist - on the portal, it should probably show the users outright that this action will overwrite the following components, do you want to proceed **(regardless of the metadata)**.

Now if a component is created by portal's `init` and then a user runs the `init` CLI, then we can do one of the following things:
1. Treat the component created by portal's `init`, the same as the component created by `dapr-component set` command or portal. In this case we continue to throw an error (CURRENT BEHAVIOR)
2. Treat the component created by portal's `init`, the same as the `dapr-component init` command. In this case we can simply overwrite it.

The current experience is in line with the portal's `init` experience (as in only overwrite components created by the tool being used). Do you think we should rather provide an alternate experience as suggested by point 2? We can achieve it by either standardizing the metadata or checking for both as you pointed earlier.

Reply:
I have updated this to the following for now.

```python
DAPRCOMPONENTINIT_SERVICEBIND_METADATA_CREATEDBY_KEY = "DCI_SB_CREATEDBY"
DAPRCOMPONENTINIT_SERVICEBIND_METADATA_CREATEDBY_VALUE = "azcli_azext_containerapp_daprutils"
```

While checking, I am only checking the presence of the key. The value can be set by CLI or Portal (in future).